conformance/lang/expressions/member-access-expr/array_member_access_expr.balt

```diff
@@ -0,0 +1,582 @@
+}
+
+Test-Case: output
+Description: Test checking expression as key expression with container expression of basic type list.
+Labels: member-access-expr, check, checkpanic, error-type, array-type, int, list-constructor-expr, union-type,
+        optional-type
+
+function init() returns error? {
+    int[] arr = [1, 2, 3, 4, 5];
+    int|error pos = 3;
+
+    io:println(arr[check pos]); // @output 4
+    io:println(arr[checkpanic pos]); // @output 4
+}
+
+Test-Case: output
+Description: Test conditional expression as key expression with container expression of basic type list.
+Labels: member-access-expr, conditional-expr, is-expr, optional-type, array-type, int, list-constructor-expr
+
+function init() {
+    int[] arr = [1, 2, 3, 4, 5];
+    int? pos = 3;
+
+    io:println(arr[pos is int ? pos : 0]); // @output 4
+}
+
+Test-Case: output
+Description: Test member access expression at module level.
+Labels: member-access-expr, module-init-var-decl, array-type, int, list-constructor-expr
+
+int[] arr = [1, 2, 3, 4, 5];
+
+int num = arr[0];
+
+function init() {
+    io:println(num); // @output 1
+}
+
+Test-Case: output
+Description: Test member access expression with container expression of basic type list as array member.
+Labels: member-access-expr, array-type, int, list-constructor-expr
+
+function init() {
+    int[] arr = [1, 2, 3, 4, 5];
+    int[] anotherArr = [arr[3], arr[0], arr[4]];
+
+    io:println(anotherArr[1]); // @output 1
+}
+
+Test-Case: output
+Description: Test member access expression with container expression of basic type list as default expression
+             for a field.
+Labels: member-access-expr, module-init-var-decl, record-type, module-type-defn, mapping-constructor-expr,
+        array-type, int, list-constructor-expr, readonly-type
+
+final int[] & readonly arr = [1, 2, 3, 4, 5];
+
+type Record record {|
+    int a = arr[2];
+|};
+
+function init() {
+    Record rec = {};
+    io:println(rec.a); // @output 3
+}
+
+Test-Case: output
+Description: Test member access expression with container expression of basic type list as defaultable
+             function parameter.
+Labels: member-access-expr, defaultable-param, function-call-expr, module-init-var-decl, array-type, int,
+        list-constructor-expr, readonly-type
+
+final int[] & readonly arr = [1, 2, 3, 4, 5];
+
+function init() {
+    io:println(getInt()); // @output 5
+}
+
+function getInt(int a = arr[4]) returns int {
+    return a;
+}
+
+Test-Case: output
+Description: Test member access expression with container expression of basic type list as expression in
+             return statement.
+Labels: member-access-expr, return-stmt, function-call-expr, array-type, int, list-constructor-expr
+
+function init() {
+    io:println(getInt([1, 2, 3, 4, 5])); // @output 5
+}
+
+function getInt(int[] arr) returns int {
+    return arr[4];
+}
+
+Test-Case: output
+Description: Test assigning a value to member access expression.
```

Comment:
Also, these rules are described under [assigment-stmt](https://ballerina.io/spec/lang/master/#assignment-stmt) where `lvexpr` is a `member-access-lvexpr`.